site/app/controllers/course/CourseMaterialsController.php

```diff
@@ -338,15 +338,59 @@ public function ajaxUploadCourseMaterialsFiles() {
                     //cannot check if there are duplicates inside zip file, will overwrite
                     //it is convenient for bulk uploads
                     if ($expand_zip == 'on' && $is_zip_file === true) {
+                        //get the file names inside the zip to write to the JSON file
+                        
                         $zip = new \ZipArchive();
-                        $res = $zip->open($uploaded_files[1]["tmp_name"][$j]);
-                        if ($res === true) {
-                            $zip->extractTo($upload_path);
-                            $zip->close();
-                            $subfiles = FileUtils::getAllFiles($upload_path, array(), true );
-                            foreach ($subfiles as $file) {
-                                $json[$file['path']] = array('checked' => '1', 'release_datetime' => $release_time  );
+-                       $res = $zip->open($uploaded_files[1]["tmp_name"][$j]);
+
+                        if(!$res){
+                            return $this->core->getOutput()->renderResultMessage("ERROR: Failed to open zip archive", false);
+                        }
+
+                        $entries = [];
+                        $disallowed_folders = [".svn", ".git", ".idea", "__macosx"];
+                        $disallowed_files = ['.ds_store'];
+                        for ($i = 0; $i < $zip->numFiles; $i++) {
+                            $entries[] = $zip->statIndex($i);
```

Comment:
Replace `statIndex` with `getNameIndex` and get rid of the indexing on `name` on line 357 and can then also get rid of lines 376-384 as well. I originally was planning on using size and/or crc values on the `$entry` for determining folder or not, but just checking the last character of the name proves reliable enough.